app/Http/Controllers/PDFController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use DOMPDF;
+use LibreNMS\Config;
+use Illuminate\Support\Facades\Route;
+use LibreNMS\Util\DynamicConfig;
+
+require '/opt/librenms/includes/init.php';
+require_once '/opt/librenms/includes/html/functions.inc.php';
```

Comment:
these are not allowed, if you need functions from the legacy code you need to port them to modern classes.
There are many examples of that inside that file, see for example;
https://github.com/librenms/librenms/blob/6c3dcfd871064337a603844158f54aa7c8a9d47c/includes/html/functions.inc.php#L61-L64

Reply:
Thanks for looking -
If I omit lines 11 & 12 these, I get this error
#Error
Call to undefined function dbFetchRows()

this was the only way i could get the database queries to work in the file referenced at line 29 
(in this case its inlcudes/html/reports/bills.pdf.inc.php)

which you say below is not allowed :(
how can I access dbFetchRows - what have I missed?

Reply:
No queries should use the dbFacile functions.  See https://laravel.com/docs/8.x/eloquent   For example for bills see App\Models\Bill